mcp_hydrolix/mcp_server.py

```diff
@@ -40,79 +112,48 @@ def list_databases():
     logger.info("Listing all databases")
     client = create_hydrolix_client()
     result = client.command("SHOW DATABASES")
-    logger.info(f"Found {len(result) if isinstance(result, list) else 1} databases")
-    return result
+
+    # Convert newline-separated string to list and trim whitespace
+    if isinstance(result, str):
+        databases = [db.strip() for db in result.strip().split("\n")]
+    else:
+        databases = [result]
+
+    logger.info(f"Found {len(databases)} databases")
+    return json.dumps(databases)
 
 
 @mcp.tool()
-def list_tables(database: str, like: str = None):
-    """List available Hydrolix tables in a database"""
+def list_tables(database: str, like: Optional[str] = None, not_like: Optional[str] = None):
+    """List available Hydrolix tables in a database, including schema, comment,
+    row count, and column count."""
     logger.info(f"Listing tables in database '{database}'")
     client = create_hydrolix_client()
-    query = f"SHOW TABLES FROM {quote_identifier(database)}"
+    query = f"SELECT database, name, engine, create_table_query, dependencies_database, dependencies_table, engine_full, sorting_key, primary_key, total_rows, total_bytes, total_bytes_uncompressed, parts, active_parts, total_marks, comment FROM system.tables WHERE database = {format_query_value(database)}"
```

Comment:
is this where we need to add `engine = 'TurbineStorage'`?

Reply:
I think that's redundant with `WHERE database = ...` in practice. This is a detail we can investigate orthogonally, for now, this code matches the content present on our main branch